lib/protobuf/new_transform.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Protobuf.NewTransform do
+  @moduledoc """
+  For structs with a `Protobuf.TransformModule` set, the `new/1` function does not get called.
+  This transform module brings back the original behavior of calling `new/1` before the value
+  is encoded.
+  """
```

Comment:
Can we add some examples here of why this matters? Like ideally an example of a call to `new` that has passes in attributes as maps.

Reply:
I added examples to `Protobuf.new/1` and we link to it from these docs.

Reply:
I think an example similar to the `WithNewTransformModule` that we have in the tests would be helpful here too.